Changelog for 2.0 (#6010)
gammapy/gammapy: docs/release-notes/v2.0.rst

  instead of ``band`` to allow exact matches
- ``cta_south`` and ``cta_north`` have been renamed as ``ctao_south`` and ``ctao_north``, respectively,
  in `~gammapy.data.observatory_locations`
Features deprecated since Gammapy v1.3 have been removed.

Documentation improvements
~~~~~~~~~~~~~~~~~~~~~~~~~~

- Documentation appearance has been updated using less custom css code, and now support a dark theme.
- The ordering of tutorials has been restructured for clarity.
- Four new tutorials have been added:

   - one explaining VERITAS analysis along with addition of VERITAS data in `$GAMMAPY_DATA` (:doc:`/tutorials/data/veritas`)
   - one explaining computation of upper limits for non-detected sources (:doc:`/tutorials/analysis-3d/non_detected_source`)
   - one explaining computation of upper limits on parameters (:doc:`/tutorials/details/parameter_ul`)
   - one introducing Bayesian inference with nested sampling (:doc:`/tutorials/details/nested_sampling_Crab`)

- The tutorial showing how to handle Fermi-LAT data (:doc:`/tutorials/data/fermi_lat`) has been rewritten to expose a full Fermi-LAT analysis.
- Code examples have been added in docstrings.

> 

Reviewer: Thank you @AtreyeeS and @Astro-Kirsty, I understand and fully agree!

- Add interp_scale to parameter scaling
- Make a distinction between `~gammapy.modeling.Parameters.free_parameters` and
  `~gammapy.modeling.Parameters.free_unique_parameters`

*gammapy.stats*

- Addition of a ``FIT_STATISTIC_REGISTRY`` to allow configuration of the statistic on a dataset.

*gammapy.utils*

- Introduce a FoVFrame in `gammapy.utils.coordinates`
- Get regions of bright stars in FoV
- Allow energy selection in `~gammapy.makers.utils.make_theta_squared_table`

Bug fixes and improvements
~~~~~~~~~~~~~~~~~~~~~~~~~~

- Improvement of the `MapDataset.peek` function to show relevant info
- Fix memory leak related to lru_cache and caching coordinates in WcsGeom
- Fix `~gammapy.estimators.TSMapEstimator` pad_width
- Fix mask handling in `~gammapy.estimators.TSMapEstimator`
- Preserve parameters links in `Models.copy()`
- Fix link label display in parameters.to_table
- Fix ``e_peak`` implementation of `~gammapy.modeling.models.SuperExpCutoffPowerLaw4FGLDR3SpectralModel` to give results compatible with Eq 21 of  https://iopscience.iop.org/article/10.3847/1538-4357/acee67
- Fix of a `~gammapy.maps.HpxGeom` simulation error

> 

Reviewer: I don't know if it is maybe too small to mention, would you think a line for #5964 could be helpful, like 
```suggestion
Added log warning for `~gammapy.data.DataStore.get_observations` in case list obs_id is not unique
```

- 100 (XX) closed issues

Summary
~~~~~~~

This is a new Long term stable release. Gammapy 2.0 will be maintained for the next two years, and we remove support for
Gammapy 1.x versions. With Gammapy 2.0, we provide analysis support for major Imaging Atmospheric Cherenkov Telescope like H.E.S.S.,
VERITAS, MAGIC, LST1 and Water Cherenkov Telesope like HAWC and SWGO, and Fermi-LAT. 

Since Gammapy 1.3, improved support is provided for Fermi-LAT analysis, and full analysis for VERITAS telescope is supported.
Bayesian inference with nester sampling is now supported.
There has been significant code cleanup, specially with respect to the documentation and the tutorial structure has
been cleaned up.

API changes
~~~~~~~~~~~

Two API-breaking changes have been introduced in this version:

- `gammapy.data.EventList.select_parameter` now takes a ``values`` and ``is_range`` argument
  instead of ``band`` to allow exact matches
- ``cta_south`` and ``cta_north`` have been renamed as ``ctao_south`` and ``ctao_north``, respectively,
  in `~gammapy.data.observatory_locations`
Features deprecated since Gammapy v1.3 have been removed.

Documentation improvements
~~~~~~~~~~~~~~~~~~~~~~~~~~

> - Documentation appearance has been updated using less custom css code, and the

Reviewer 2: Docs support a dark scheme now...

.. include:: ../references.txt

.. _gammapy_2p0_release:

2.0 (August XX, 2025)
----------------

- Released August XX, 2025
- 25 contributors
- 235 (XX)  pull requests since v1.3 (not all listed below)
- 100 (XX) closed issues

Summary
~~~~~~~

This release introduces a number of performance improvements, new features and bug fixes.
Improved support is provided for Fermi-LAT analysis, and full analysis for VERITAS telescope is supported.
Bayesian inference with nester sampling is now supported.
There has been significant code cleanup, specially with respect to the documentation and the tutorial structure has

> been cleaned up. cython code for computation of fit statistics is now replaced by jit code.
> Dependence on `lru_cache` is removed.

Reviewer 3: ```suggestion
been cleaned up.
```
I would not mention cython and lru_cache here but only on the details. And cython is still the default using jit is optional.

API changes
~~~~~~~~~~~

Two API-breaking changes have been introduced in this version:

- `~gammapy.EventList.select_parameter` now takes a ``values`` and ``is_range`` argument instead of ``band`` to allow exact matches
- ``cta_south`` and `cta_north` have been renamed as ``ctao_south`` and ``ctao_north``, resp. in `~gammapy.data.observatory_locations`
Features deprecated since Gammapy v1.3 have been removed.

Infrastructure
~~~~~~~~~~~~~~

- TODO

Documentation improvements
~~~~~~~~~~~~~~~~~~~~~~~~~~

- Documentation appearance has been updated using less custom css code,
and the ordering of tutorials restructured for clarity.
- Four new tutorials have been added:
   - one explaining VERITAS analysis along with addition of VERITAS data in `$GAMMAPY_DATA`
   - one explaining computation of upper limits for non-detected sources
   - one explaining computation of upper limits on parameters
   - one introducing Bayesian inference with nested sampling

> - The tutorial exposing FERMI-LAT analysis has been rewritten to expose the support for FERMI analysis

Reviewer 4: ```suggestion
- The tutorial exposing Fermi-LAT analysis has been rewritten to expose the support for Fermi analysis
```

> - The tutorial exposing FERMI-LAT analysis has been rewritten to expose the support for FERMI analysis

Reviewer 3: ```suggestion
- The tutorial showing how to handle Fermi-LAT data has been rewritten to expose a full Fermi-LAT analysis.
```

- Code examples have been added in docstrings.

New features
~~~~~~~~~~~~

*gammapy.analysis*

> - Improvement of the High Level Interface using the `Analysis` class

Reviewer 4: ```suggestion
- Improvement of the High Level Interface using the `~gammapy.analysis.Analysis` class
```

> *gammapy.analysis*
> 
> - Improvement of the High Level Interface using the `Analysis` class

Reviewer 3: ```suggestion
```
Not really there yet, the new gammapy.workflow is still not public api.

*gammapy.catalog*

- Add pulse profiles for Fermi 3PC catalog

*gammapy.data*

- ￼Helper method for observation selection at Datastore level, specially for WCD

*gammapy.datasets*

> - Add `Datasets.to_asimov_datasets`

Reviewer 3: ```suggestion
```
`_to_asimov_datasets` and `_to_asimov_dataset` are not public.
We only use them internally to compute sensitivity.

- Add a helper function to create a global dataset from a list of datasets

*gammapy.estimators*

> - Add optional sensitivity computation on TSMapEstimator

Reviewer 3: ```suggestion
- Add optional sensitivity computation on TSMapEstimator and FluxPointsEstimator
```

*gammapy.makers*

- Added spatial model to `gammapy.makers.FoVBackgroundMaker` init

> - Add a `FermiDatasetMaker`

Reviewer 3: ```suggestion
- Add a `FermipyDatasetsReader` convert Fermi-LAT maps from Fermipy into a gammapy Datasets.
```

*gammapy.modeling*

> - Support for weighted likelihood as implemented in, eg: FERMI-LAT

Reviewer 3: ```suggestion
- Support for pixel-wise weighted likelihood as implemented in Fermi-LAT science tools
```

- Support for nested sampling
- Addition of a `LogUniformPrior`

> - Add ``datasets_names argument`` to ``create_fermi_isotropic_diffuse_model``

Reviewer 3: ```suggestion
- Implement a more robust error propagation method using sampling
- Add ``datasets_names argument`` to ``create_fermi_isotropic_diffuse_model``
```

- Add interp_scale to parameter scaling
- Make a distinction between `free_parameters` and `free_unique_parameters`

*gammapy.stats*

- Addition of a `FIT_STATISTIC_REGISTRY` to allow configuration of the statistic on a dataset.

*gammapy.utils*

- Introduce a FoVFrame in `gammapy.utils.coordinates`
- Get regions of bright stars in FoV
- Allow energy selection in `gammapy.makers.utils.make_theta_squared_table`
- Introduction of a ``PolygonPointsSkyRegion``

Bug fixes and improvements
~~~~~~~~~~~~~~~~~~~~~~~~~~

- Improvement of the `MapDataset.peek` function to show relevant info

> - Fix reading of `TemplateSpatialModels` written with Gammapy v1.1 and earlier.

Reviewer 3: ```suggestion
```
It's not completely fix as it just skip reading the covariance.

- Fix memory leak related to lru_cache

> - Fix TSmapEstimator pad_width

Reviewer 3: ```suggestion
- Fix TSmapEstimator pad_width
- Fix mask handling in TSMapEstimator
```

- Fix parameters links after models copy
- Fix link label display in parameters.to_table
- Correct small bug in PSFMap
- Fix e_peak implementation of SuperExpCutoffPowerLaw4FGLDR3SpectralModel
- Fix of a HpXGeom simulation error

Contributors
~~~~~~~~~~~~
- Leander Schlegel
- Luis Barrios-Jiménez
- Quentin Remy
- Tomas Bylund
- Tora T H Arnesen
- Anushka Bhatt
- Bruno Khélifi
- REGEARD Maxime
- Daniel Morcuende
- Robert Daniel Parsons
- Mireia Nievas-Rosillo
- Fabio PINTORE
- Atreyee Sinha
- lgreaux
- Arnau Aguasca-Cabot
- Fabio Acero

> - npigoux

Reviewer 5: ```suggestion
- Natthan Pigoux
```

- [#5936] Update of the user guide documentation of `Modelling and Fitting` (Bruno Khélifi)
- [#5929] Add note on PR doc build access (Kirsty Feijen)
- [#5927] Add mask plot for spectral (Kirsty Feijen)
- [#5923] simple fix to problem with EventList printing (Tomas Bylund)
- [#5918] Adding VERITAS tutorial (Samantha Wong)
- [#5915] Fix parameters links after models copy (Quentin Remy)
- [#5914] Modify sampler to handle linked params in models (Fabio Acero)
- [#5913] Add a function to retrieve free and unique parameters (Fabio Acero)
- [#5912] moving linkcheck from ci to a separate workflow (Hanna Stapel)
- [#5911] Introducing generic reproject_irf_on_geom functions using FoVFrame classes (Régis Terrier)
- [#5888] Fix gammapy.catalog to support astropy v7.1 (Quentin Remy)
- [#5884] Rm setting log levels in modules (Tobi Preis)
- [#5880] Serialization of GTI extension is not correct in the event sampling tutorial (Fabio PINTORE)
- [#5870] Add method option to Irf.integrate_log_log (Régis Terrier)
- [#5860] Introduce a FoVFrame in gammapy.utils.coordinates (Régis Terrier)
- [#5855] Utility to filter a table parsing string of  logical operations (Quentin Remy)

> - [#5797] Sampling tutorial (Fabio Acero)

Reviewer 3: ```suggestion
- [#5797] Sampling tutorial (Fabio Acero)
- [#5786] More robust error propagation approach using sampling  (Quentin Remy)
```

- [#5766] Modification of peek() in MapDataset to show more information (Mireia Nievas-Rosillo)
- [#5760] Added spatial model to FoVBackgroundMaker init (Katharina Egg)
- [#5756] 3PC pulse profile (REGEARD Maxime)
- [#5748] Replace cython code by jitted code (Quentin Remy)
- [#5744] Fix of a HpXGeom simulation error (Mitsunari-Takahashi)
- [#5740] Apply mask in MapDataset.plot_residuals_spectral (Quentin Remy)
- [#5733] Remove the unneeded css code for the dropdowns (Kirsty Feijen)
- [#5731] fix: install sherpa using conda instead of pip in dev env (npigoux)
- [#5722] Color-blind friendly links (Bruno Khélifi)
- [#5719] Limit misusage of `to_region_map_dataset` (Quentin Remy)
- [#5716] Documentation appearance (Kirsty Feijen)
- [#5715] Modified Polygon region for use as ON region (Robert Daniel Parsons)
- [#5703] Make theta squared distribution selected in energy (Fabio PINTORE)
- [#5699] Add `mask` option to `plot_distribution()` (Katharina Egg)

> - [#5696] Introducing FitStatistic (Régis Terrier)

Reviewer 3: ```suggestion
- [#5696] Introducing FitStatistic (Régis Terrier)
- [#5690] Fix and cleanup mask handling in TSMapEstimator (Quentin Remy)
```

- ￼Helper method for observation selection at Datastore level, specially for WCD

*gammapy.datasets*

- Add a helper function to create a global dataset from a list of datasets

*gammapy.estimators*

- Add optional sensitivity computation on TSMapEstimator and FluxPointsEstimator

*gammapy.makers*

- Added spatial model to `gammapy.makers.FoVBackgroundMaker` init
- Add a `FermipyDatasetsReader` to convert Fermi-LAT maps from Fermipy into a gammapy Datasets.

*gammapy.modeling*

- Support for pixel-wise weighted likelihood as implemented in Fermi-LAT science tools
- Support for nested sampling
- Addition of a `LogUniformPrior`
- Implement a more robust error propagation method using sampling
- Add ``datasets_names argument`` to ``create_fermi_isotropic_diffuse_model``
- Add interp_scale to parameter scaling
- Make a distinction between `free_parameters` and `free_unique_parameters`

*gammapy.stats*

- Addition of a `FIT_STATISTIC_REGISTRY` to allow configuration of the statistic on a dataset.

*gammapy.utils*

- Introduce a FoVFrame in `gammapy.utils.coordinates`
- Get regions of bright stars in FoV
- Allow energy selection in `gammapy.makers.utils.make_theta_squared_table`

> - Introduction of a ``PolygonPointsSkyRegion``

Author: removed

.. include:: ../references.txt

.. _gammapy_2p0_release:

2.0 (August XX, 2025)
----------------

- Released August XX, 2025
- 25 contributors
- 235 (XX)  pull requests since v1.3 (not all listed below)
- 100 (XX) closed issues

Summary
~~~~~~~

This is a new Long term stable release. Gammapy 2.0 will be maintained for the next two years, and we removes support for

> Gammapy 1.x versions. With Gammapy 2.0, we provide analysis support for major Imaging Atmospheric Cherenkov Telescope like H.E.S.S.,

Reviewer 2: Minor: there are these random line breaks, please remove consistently. Line breaks in md/rst documents are typically used to separate paragraphs. The actual line breaks are then introduced by presentation in the browser.

There has been significant code cleanup, specially with respect to the documentation and the tutorial structure has
been cleaned up.

API changes
~~~~~~~~~~~

Two API-breaking changes have been introduced in this version:

- `gammapy.data.EventList.select_parameter` now takes a ``values`` and ``is_range`` argument
  instead of ``band`` to allow exact matches
- ``cta_south`` and ``cta_north`` have been renamed as ``ctao_south`` and ``ctao_north``, respectively,
  in `~gammapy.data.observatory_locations`
Features deprecated since Gammapy v1.3 have been removed.

Infrastructure
~~~~~~~~~~~~~~

- removed dependency on ``sphinx-astropy``
- utilise OpenAstronomy in our release workflow to build wheels
- added extra workflows:

   - scheduled run for doctests in rst and examples
   - scheduled link check (separated from ci)

- added the docs build to PRs
- support of astropy v7.1
- allow skipping of ci actions through labels: ``skip-docs``, ``skip-tests``

Documentation improvements
~~~~~~~~~~~~~~~~~~~~~~~~~~

- Documentation appearance has been updated using less custom css code, and the
  ordering of tutorials restructured for clarity.
- Four new tutorials have been added:

>    - one explaining VERITAS analysis along with addition of VERITAS data in `$GAMMAPY_DATA`

Reviewer 2: Add links to the actual tutorials.

- Support for nested sampling
- Addition of a `~gammapy.modeling.models.LogUniformPrior`
- Implement a more robust error propagation method using sampling
- Add ``datasets_names`` argument to `~gammapy.modeling.models.create_fermi_isotropic_diffuse_model`
- Add interp_scale to parameter scaling
- Make a distinction between `~gammapy.modeling.Parameters.free_parameters` and
  `~gammapy.modeling.Parameters.free_unique_parameters`

*gammapy.stats*

- Addition of a ``FIT_STATISTIC_REGISTRY`` to allow configuration of the statistic on a dataset.

*gammapy.utils*

- Introduce a FoVFrame in `gammapy.utils.coordinates`
- Get regions of bright stars in FoV
- Allow energy selection in `~gammapy.makers.utils.make_theta_squared_table`
- Introduction of a ``PolygonPointsSkyRegion``

Bug fixes and improvements
~~~~~~~~~~~~~~~~~~~~~~~~~~

- Improvement of the `MapDataset.peek` function to show relevant info
- Fix memory leak related to lru_cache
- Fix `~gammapy.estimators.TSMapEstimator` pad_width
- Fix mask handling in `~gammapy.estimators.TSMapEstimator`
- Fix parameters links after models copy
- Fix link label display in parameters.to_table
- Correct small bug in PSFMap

> - Fix ``e_peak`` implementation of `~gammapy.modeling.models.SuperExpCutoffPowerLaw4FGLDR3SpectralModel`

Reviewer 2: Again being a bit more specific would be good.

   - one explaining computation of upper limits for non-detected sources
   - one explaining computation of upper limits on parameters
   - one introducing Bayesian inference with nested sampling

> - The tutorial showing how to handle Fermi-LAT data has been rewritten to expose a full Fermi-LAT analysis.

Reviewer 6: There is an issue with the rendering of the sublist
